File: python/2020/june/reverse-string.py
from typing import List


class Solution:
    def reverse_string(self, s: List[str]) -> None:
        self.helper(s, 0, len(s) - 1)

    def helper(self, s, l, r):
        if l == r or l - r == 1:
            return
        s[l], s[r] = s[r], s[l]
        self.helper(s, l + 1, r - 1)

File: go/2020/june/reverse-string.go
package june

func reverseString(s []rune) {
	helper(s, 0, len(s)-1)
}

func helper(s []rune, l int, r int) {
	if l == r || l-r == 1 {
		return
	}
	s[l], s[r] = s[r], s[l]
	helper(s, l+1, r-1)
}
